### PR TITLE
docs: updates to overview & ReadMe pages

### PR DESCRIPTION
**Pull request by NeethuES-intel**
## PR Checklist

- [X] Added label to the Pull Request for easier discoverability and search
- [X] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [ ] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [X] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
Updated Overview & ReadMe pages to include more details on the use case

## Test Instructions if applicable

- `make serve-docs`
- Page should render correctly

### CODE REVIEW

File: docs_src/index.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Overview
 
+As Computer Vision is becoming more and more popular especially, for industrial & retail use cases, development and deployment of these solutions are becoming more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below -
```
Comment (jim-wang-intel) [inline docs_src/index.md:5]:
```suggestion
As Computer Vision is becoming more and more popular, especially for industrial & retail use cases, development and deployment of these solutions are becoming more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below -
```

File: docs_src/index.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
...
+
+[![Vision Data Flow](./images/vision-data-flow.jpg)](./images/vision-data-flow.jpg)
+
+Automated self-checkout solutions are complex, and retailers, independent software vendors (ISVs), and system integrators (SIs) require a good understanding of hardware and software, the costs involved in setting up and scaling the system, and the configuration that best suits their needs. Vision workloads are significantly larger and require systems to be architected, built, and deployed with several considerations. Hence, a set of ingredients needed to create an automated self-checkout solution is necessary. More details on this use case are available on the [Intel Developer Focused Webpage](https://www.intel.com/content/www/us/en/developer/articles/reference-implementation/automated-self-checkout.html) and on this [LinkedIn Blog](https://www.linkedin.com/pulse/retail-innovation-unlocked-open-source-vision-enabled-mohideen/)
```
Comment (ejlee3) [inline docs_src/index.md:9]:
```suggestion
Automated self-checkout solutions are complex; retailers, independent software vendors (ISVs), and system integrators (SIs) require a good understanding of hardware and software, the costs involved in setting up and scaling the system, and the configuration that best suits their needs. Vision workloads are significantly larger and require systems to be architected, built, and deployed with several considerations. Hence, automated self-checkout solutions require customizability to fit the use case. More details are available on the [Intel Developer Focused Webpage](https://www.intel.com/content/www/us/en/developer/articles/reference-implementation/automated-self-checkout.html) and on this [LinkedIn Blog](https://www.linkedin.com/pulse/retail-innovation-unlocked-open-source-vision-enabled-mohideen/)
```

**Review by ejlee3 — CHANGES_REQUESTED** [review]

File: docs_src/index.md [hunk 3]
```diff
@@ -2,6 +2,12 @@
 
 ## Overview
 
+As Computer Vision is becoming more and more popular, especially for industrial & retail use cases, development and deployment of these solutions are becoming more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below -
```
Comment (ejlee3) [inline docs_src/index.md:5]:
```suggestion
As Computer Vision becomes more and more mainstream, especially for industrial & retail use cases, development and deployment of these solutions becomes more challenging. Vision workloads are large and complex and need to go through many stages as shown in the figure below.
```

File: docs_src/index.md (hunk 2 above)
Comment (NeethuES-intel) [reply docs_src/index.md:9]: these are mostly from the blog/developer page, won't make much changes.

**Review by ejlee3 — APPROVED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**